pkg/common/cns-lib/volume/manager.go

```diff
@@ -595,8 +595,10 @@ func (m *defaultManager) waitOnTask(csiOpContext context.Context,
 	}
 	ch := make(chan TaskResult)
 	err := m.listViewIf.AddTask(csiOpContext, taskMoRef, ch)
-	if err != nil {
+	if errors.Unwrap(err) == ErrListViewAddition {
 		return nil, logger.LogNewErrorf(log, "%s. err: %v", listviewAdditionError, err)
+	} else if err != nil {
+		return nil, err
```

Comment:
fyi, we need to differentiate two types of errors here - 

1. `ErrListViewAddition` refers to any errors returned by vpxd when we try to add a task to ListView. 
2. The other error is the `ManagedObjectNotFound` fault returned as a soapfault for inactive aka unresolved tasks.

Reply:
can we add a comment saying the error in else case will be ManagedObjectNotFound that gets handled in the caller for expired task case ?

Reply:
Added.